Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -124,7 +124,7 @@ void TRDGlobalTracking::run(ProcessingContext& pc)
   for (int iTrk = 0; iTrk < mChainTracking->mIOPtrs.nTracksTPCITSO2; ++iTrk) {
     const auto& trkITSTPC = mChainTracking->mIOPtrs.tracksTPCITSO2[iTrk];
     GPUTRDTrack trkLoad(trkITSTPC, mTPCVdrift);
-    if (mTracker->LoadTrack(trkLoad, -1, nullptr, -1, iTrk)) {
+    if (mTracker->LoadTrack(trkLoad, trkITSTPC.getRefTPC())) {
```

Comment:
Actually, I would not use getRefTPC() here, this should be the id of the reference track used for the matching, and if that was a TPC-ITS track, I would point to that one.

Reply:
Not sure if that is better. This track will end as `MATCHTRD_GLO`, i.e. as ITS-TPC-TRD track and therefore it is clear that it is from a TPC-ITS seed. If I change it to
```suggestion
    if (mTracker->LoadTrack(trkLoad, iTrk)) {
```
then I don't know after the tracking is done which seeding type the TRD tracks had. Although in `GPUTRDTrack::mIsFindable` there is an unused bit which I could use to distinguish between TPC-only and ITS-TPC. Eventually I thing I will use the RecoContainer logic for that.

Reply:
But for the further processing one needs to know which TPCITS track it belongs to. And the mapping it TPCITS --> TPC, i.e. if you store the TPC id, that would require a revert search.

I think we should use the GlobalTrackID. We can still store it as the 32 bit integer in the GPUTRDTrack, since GlobalTrackID is an abstractRef, which is just 32 bit. You can just add a getter and setter in the TrackTRD, which converts it forth and back. (with a `static_assert(sizeof(GlobalTrackId) == sizeof(whatever we use))')

Then you can get the seedType via the GlobalTrackId::getSource()

Reply:
Sounds good, I will try that. I just realized now that since `class GlobalTrackID : public AbstractRef<25, 5, 2>` the `AbstractRef::Base_t` is a 25 + 5 + 2 = 32 bit integer. That is not super obvious when one only takes a very quick look at those classes

Reply:
But now the `TrackTPCITS` has two `GlobalTrackID`s, one for the ITS track and one for the TPC track. There is no `GlobalTrackID` for the ITS-TPC match, or is there?

Reply:
You just use `GlobalTrackId{iTrk, GlobalTrackID::ITSTPC}` to construct one, this will refer to the ITS-TPC track. If the user wants to go to the TPC track, he has to follow the ids.

Reply:
Same for the TPC track, just `GlobalTrackId{iTrk, GloballTrackID::TPC}`

Reply:
Is https://github.com/AliceO2Group/AliceO2/pull/6107/commits/aa3b2a368bbfef8473bc29e0d9946f2f9d31e909  what you meant?